cors.go

```diff
@@ -79,7 +184,17 @@ func AllowedHeaders(headers []string) CORSOption {
 // AllowedMethods ...
 func AllowedMethods(methods []string) CORSOption {
 	return func(ch *cors) error {
-		ch.allowedMethods = methods
+		for _, v := range methods {
+			normalizedMethod := strings.ToUpper(strings.TrimSpace(v))
+			if normalizedMethod == "" {
+				continue
+			}
+
+			if !ch.isMatch(normalizedMethod, ch.allowedMethods) {
+				ch.allowedHeaders = append(ch.allowedHeaders, normalizedMethod)
```

Comment:
In this instance we probably want to override the allowed methods completely with the user's preferences—i.e. they may only want to allow `GET` and not the default set of `GET, HEAD, POST`.

Double check what the standard says here however.